packages/pyright-internal/src/tests/semanticTokensProvider.test.ts

```diff
@@ -142,6 +142,47 @@ if (process.platform !== 'win32' || !process.env['CI']) {
             { type: 'class', modifiers: ['defaultLibrary', 'builtin'], start: 154, length: 16 }, // EnvironmentError
         ]);
     });
+
+    test('decorators', () => {
+        const result = semanticTokenizeSampleFile('decorators.py');
+        expect(result).toStrictEqual([
+            // TODO: Fix duplicates
+            { type: 'namespace', modifiers: [], start: 7, length: 11 }, // dataclasses
+            { type: 'namespace', modifiers: [], start: 24, length: 11 }, // dataclasses
+            { type: 'function', modifiers: [], start: 43, length: 9 }, // dataclass
+            { type: 'function', modifiers: [], start: 43, length: 9 }, // dataclass
+            { type: 'namespace', modifiers: [], start: 60, length: 9 }, // functools
+            { type: 'namespace', modifiers: [], start: 75, length: 6 }, // typing
+            { type: 'function', modifiers: [], start: 89, length: 5 }, // final
+            { type: 'function', modifiers: [], start: 89, length: 5 }, // final
+
+            { type: 'class', modifiers: ['definition'], start: 115, length: 1 }, // A
+            { type: 'decorator', modifiers: [], start: 96, length: 1 }, // @
+            { type: 'decorator', modifiers: [], start: 97, length: 9 }, // dataclass
+            { type: 'function', modifiers: [], start: 97, length: 9 },
```

Comment:
this doesn't take priority over the decorator one does it?

Reply:
No, I think. Honestly I don't know if the priority of the token is based on the order the tokens are sent, the color scheme or the editor configuration. When I tested it in Neovim it seemed that  it gave lower priority to the decorator no matter the order.

Reply:
as in, it incorrectly shows the decorator as a function instead of a decorator?

Reply:
Yes, it shows the decorator as a function, but I'm not sure if that is just my configuration. The editor still receives both tokens and applies them, but gives lower priority to the decorator. Might be worth testing it in VS Code, but I don't use it.

Reply:
looks like by default in vscode decorators have the same colour as functions, so class decorators like `staticmethod` now look like functions when used as decorators, which i believe is the intended behavior.

ideally `visitName` shouldn't be touching names that have already been given a token by another visitor method (which is also what's causing those duplicate tokens i think), but it hasn't caused any issues yet so i guess it's fine as is for now.